### DISCUSSION TITLE
Obsługa wielu ACI

### DISCUSSION THREAD

**Discussion by arturmeller** [AzureCommunityPL/group#144] [Pomoc]
Cześć, 
Mam temat zarządzania wieloma instancjami ACI, wieloma to znaczy w setkach. Chodzi o świadczenie usług dla klientów, którzy muszą być odseparowani od siebie. Co proponujecie do ogarnięcia takiej sytuacji?
Próbowałem z Portainerem, ale on nie może odczytać istniejących ACI i przy deploymencie kontenera tworzy oddzielną instancje ACI per kontener. Nie znalazłem możliwości powołania kilku kontenerów w jednym ACI.
Co można użyć żeby kontrolować te setki ACI w jednym miejscu?

**Comment by kaluzaaa** [comment]
1. AKS i Virtual Kubelet
1. REST API i własna nakładka

**Reply by arturmeller** [reply]
Pierwsza opcja odpada, musi być separacja (bo KNF). Druga to zło konieczne :-(

**Reply by kaluzaaa** [reply]
@arturmeller bo KNF to zła odpowiedź. Zależy naprawdę co rozumiesz pod nazwą separacja i jaka ona musi być ;-)

**Reply by mgrabarz** [reply]
Rozkminy takie jak ostatnio u naszego klienta @kaluzaaa ;)

**Comment by piotrgwiazda** [comment]
Co do ogrania tematu ACI to super sobie radzi z tym Terraform.
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/container_group

Możesz je robić w masowo w pętli (mechanizm count).

Co do KNF to rozumiem, że opierasz nie na tym:
"Historically, containers have offered application dependency isolation and
resource governance but have not been considered sufficiently hardened for
hostile multi-tenant usage. Azure Container Instances guarantees your
application is as isolated in a container as it would be in a VM"

Do ogrania tego:
"4.Informacje o mechanizmach
izolacji zasobów używanych
do świadczenia usług chmury
obliczeniowej, w tym informacje
o incydentach bezpieczeństwa
związanych z naruszeniem
mechanizmów izolacji"



wt., 10 lis 2020, 17:23 użytkownik Marek Grabarz <notifications@github.com>
napisał:

**Reply by kaluzaaa** [reply]
@piotrgwiazda slowo bank nie oznacza, że to wszystkie komunikaty i rekomendacje trzeba stosować ;-) to jest per przypadek kiedy spełnia kryteria, że musi :-)

Co do terraform to nie nadaje się z mojego doświadczenia do zarządzania cyklem życia jak w przypadku pytania.
